src/linear_algebra/linear_pmap.lean

```diff
@@ -419,3 +419,39 @@ def coprod (f : linear_pmap R E G) (g : linear_pmap R F G) :
 rfl
 
 end linear_pmap
+
+/-! ### Graph -/
+section graph
+
+namespace linear_pmap
+
+/-- The graph of a `linear_pmap` viewed as a submodule on `E × F`. -/
+def graph (f : linear_pmap R E F) : submodule R (E × F) :=
+f.to_fun.graph.map (f.domain.subtype.prod_map linear_map.id)
+
+lemma mem_graph_iff' (f : linear_pmap R E F) {x : E × F} :
+  x ∈ graph f ↔ ∃ (y : f.domain), (↑y, f y) = x :=
+by simp[graph]
+
+@[simp] lemma mem_graph_iff (f : linear_pmap R E F) {x : E × F} :
+  x ∈ graph f ↔ ∃ (y : f.domain), x.fst = ↑y ∧ x.snd = f y :=
+begin
+  rw mem_graph_iff',
+  split; intro h; cases h with y h; use y,
```

Comment:
I would have thought you could get there faster with `cases x` and then `simp_rw mk_eq_mk`

Reply:
I agree that there should be a better proof, but I don't understand what you are suggesting. Using `cases` I got this proof, which is not really shorter, but more understandable:
````
  rw mem_graph_iff',
  cases x,
  simp only [prod.mk.inj_iff],
  split; intro h; cases h with y h; use y;
  exact ⟨h.1.symm, h.2.symm⟩,
````